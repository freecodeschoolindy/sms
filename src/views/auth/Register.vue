<template>
  <form class="grid-x">
    <div class="cell small-4 grid-x grid-padding-x">
      <div class="cell">
        <h2>Register Page</h2>
      </div>

      <div class="cell">
        <label>
          Github Username:
          <input v-model="github" type="text" placeholder="Github Username" />
        </label>
      </div>
      <div class="cell">
        <label>
          Email:
          <input v-model="email" type="text" placeholder="Email" />
        </label>
      </div>
      <div class="cell">
        <label>
          Select Student Status:
          <select v-model="studentStatus">
            <option value="current">I am currently a student</option>
            <option value="former">I am a former student</option>
            <option value="waitlist">I want to be on the wait list to become a student</option>
            <option value="other">Other</option>
          </select>
        </label>
      </div>
      <div class="cell">
        <label>
          First Name:
          <input v-model="firstName" type="text" placeholder="First Name" />
        </label>
      </div>
      <div class="cell">
        <label>
          Last Name:
          <input v-model="lastName" type="text" placeholder="Last Name" />
        </label>
      </div>
      <div class="cell">
        <label>
          Preferred Name:
          <input v-model="preferredName" type="text" placeholder="Preferred Name" />
        </label>
      </div>
      <div class="cell">
        <button class="button" @click="registerUser">Submit</button>
      </div>
    </div>
  </form>
</template>

<script>
export default {
  data: function () {
    return {
      github: '',
      email: '',
      firstName: '',
      lastName: '',
      preferredName: '',
      studentStatus: ''
    };
  },
  methods: {
    registerUser: function () {

    }
  }
};
</script>
