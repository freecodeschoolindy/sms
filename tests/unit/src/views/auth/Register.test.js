import Register from '@/views/auth/Register.vue';

import testHelpers from '@@/testHelpers.js';

describe('Register.vue', () => {
  let localVue;
  let store;
  let setup;

  let permissions = ['student'];

  beforeEach(() => {
    setup = testHelpers.setupTestVariables(permissions);
    localVue = setup.localVue;
    store = setup.store;
  });

  test('Renders correctly', async () => {
    const wrapper = await testHelpers.shallowMountWrapper(Register, {
      localVue,
      store
    });

    expect(wrapper)
      .toMatchSnapshot();
  });
});
