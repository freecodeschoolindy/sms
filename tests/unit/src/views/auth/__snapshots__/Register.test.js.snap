// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register.vue Renders correctly 1`] = `
<form class="grid-x">
  <div class="cell small-4 grid-x grid-padding-x">
    <div class="cell">
      <h2>Register Page</h2>
    </div>
    <div class="cell">
      <label>
        Github Username:
        <input type="text" placeholder="Github Username">
      </label>
    </div>
    <div class="cell">
      <label>
        Email:
        <input type="text" placeholder="Email">
      </label>
    </div>
    <div class="cell">
      <label>
        Select Student Status:
        <select>
          <option value="current">I am currently a student</option>
          <option value="former">I am a former student</option>
          <option value="waitlist">I want to be on the wait list to become a student</option>
          <option value="other">Other</option>
        </select>
      </label>
    </div>
    <div class="cell">
      <label>
        First Name:
        <input type="text" placeholder="First Name">
      </label>
    </div>
    <div class="cell">
      <label>
        Last Name:
        <input type="text" placeholder="Last Name">
      </label>
    </div>
    <div class="cell">
      <label>
        Preferred Name:
        <input type="text" placeholder="Preferred Name">
      </label>
    </div>
    <div class="cell">
      <button class="button">Register with Github</button>
    </div>
  </div>
</form>
`;
